corehq/apps/reports/filters/api.py

```diff
@@ -123,12 +124,18 @@ def data_sources(self):
         # when searcing for custom locations search limit to just locations
         if self.custom_locations_search():
             return [(self.get_locations_size, self.get_locations)]
-        return [
-            (self.get_static_options_size, self.get_static_options),
-            (self.get_groups_size, self.get_groups),
-            (self.get_locations_size, self.get_locations),
-            (self.get_users_size, self.get_users),
-        ]
+        elif self.request.can_access_all_locations:
+            return [
+                (self.get_static_options_size, self.get_static_options),
+                (self.get_groups_size, self.get_groups),
+                (self.get_locations_size, self.get_locations),
+                (self.get_users_size, self.get_users),
+            ]
+        else:
+            return [
+                (self.get_locations_size, self.get_locations),
+                (self.get_users_size, self.get_users),
+            ]
```

Comment:
nit: i preferred how it was done earlier with appending whats needed according to access. One need not compare between two lists to figure out which data sources are different.

```
        sources = []
        if self.custom_locations_search():
            return [(self.get_locations_size, self.get_locations)]
        elif self.request.can_access_all_locations:
            sources.append((self.get_static_options_size, self.get_static_options))
            sources.append((self.get_groups_size, self.get_groups))
            
        sources.append((self.get_locations_size, self.get_locations))
       # appending this in the end to avoid long list of users delaying
       # locations, groups etc in the list on pagination
       sources.append((self.get_users_size, self.get_users))
       return sources
```

Reply:
it's a bit of repetition, but I like this way more.  I think of it as a declarative spec - "restricted users get these data sources, regular users get these ones".  It could certainly get messy if there were a bunch of different conditionals being applied, but as-is it kinda just lays it all out in one place.